ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/image/ImageComponentView.kt

```diff
@@ -0,0 +1,123 @@
+                modifier = Modifier
+                    .size(style.size)
+                    .applyIfNotNull(style.colorStyle) { colorStyle ->
+                        drawWithCache {
+                            onDrawWithContent {
+                                drawContent()
+                                when (colorStyle) {
+                                    is ColorStyle.Solid -> drawRect(colorStyle.color)
+                                    is ColorStyle.Gradient -> drawRect(colorStyle.brush)
+                                }
+                            }
+                        }
+                    },
+                placeholderUrlString = style.urls.urlsForCurrentTheme.webpLowRes.toString(),
+                contentScale = style.contentScale,
+            )
+        }
+    }
+}
+
+@Preview
+@Composable
+private fun ImageComponentView_Preview_Default() {
+    Box(modifier = Modifier.background(ComposeColor.Red)) {
+        ImageComponentView(
+            style = previewImageComponentStyle(),
+        )
+    }
+}
+
+@Suppress("MagicNumber")
+@Preview
+@Composable
+private fun ImageComponentView_Preview_LinearGradient() {
+    Box(modifier = Modifier.background(ComposeColor.Red)) {
+        ImageComponentView(
+            style = previewImageComponentStyle(
+                colorStyle = ColorStyle.Gradient(
+                    Brush.verticalGradient(
+                        Pair(0f, ComposeColor(Color.parseColor("#88FF0000"))),
+                        Pair(0.5f, ComposeColor(Color.parseColor("#8800FF00"))),
+                        Pair(1f, ComposeColor(Color.parseColor("#880000FF"))),
+                    ),
+                ),
+            ),
+        )
+    }
+}
+
+@Suppress("MagicNumber")
+@Preview
+@Composable
+private fun ImageComponentView_Preview_RadialGradient() {
+    Box(modifier = Modifier.background(ComposeColor.Red)) {
+        ImageComponentView(
+            style = previewImageComponentStyle(
+                colorStyle = ColorStyle.Gradient(
+                    Brush.radialGradient(
+                        Pair(0f, ComposeColor(Color.parseColor("#88FF0000"))),
+                        Pair(0.5f, ComposeColor(Color.parseColor("#8800FF00"))),
+                        Pair(1f, ComposeColor(Color.parseColor("#880000FF"))),
+                    ),
+                ),
+            ),
+        )
+    }
+}
+
+@Suppress("LongParameterList")
+@Composable
+private fun previewImageComponentStyle(
+    url: URL = URL("https://sample-videos.com/img/Sample-jpg-image-5mb.jpg"),
+    lowResURL: URL = URL("https://assets.pawwalls.com/954459_1701163461.jpg"),
+    visible: Boolean = true,
+    size: Size = Size(width = SizeConstraint.Fixed(400u), height = SizeConstraint.Fixed(400u)),
+    contentScale: ContentScale = ContentScale.Crop,
+    colorStyle: ColorStyle? = null,
+) = ImageComponentStyle(
+    visible = visible,
+    themeImageUrls = ThemeImageUrls(light = ImageUrls(url, url, lowResURL, 200u, 200u)),
```

Comment:
I'm not sure if the sizes of the images are affecting anything... Not according to my tests 🤔

Reply:
We can use the width and height to already give the image view the appropriate size before the image is downloaded. That way the layout won't "jump" when the image arrives. 

Since width and height are part of `ThemeImageUrls`, which is provided by `ImageComponentStyle`, I think we can remove the separate `size` from `ImageComponentStyle`.

Reply:
Ohhh ok that makes a lot of sense 😅  Thanks!

Reply:
Hmm as I was thinking about this, how would that work if we want the image to "Fill" or "Fit" the screen? In the `ThemeImageUrls` there is no way to pass if we want one of those two right?  I think passing the specific width/height only makes sense if we want a "fixed" size...

Reply:
Summary of offline conversation:
- We cannot remove the `size` as that indicates the size of the image view, not the image itself. 
- The `ImageUrls` width/height come into play when the image view is set to `Fit`.